ecl/hql/hqlfold.cpp

```diff
@@ -1335,8 +1346,486 @@ IValue * foldExternalCall(IHqlExpression* expr, unsigned foldOptions, ITemplateC
+    assertex(outofline->getOperator() == no_outofline);
+    IHqlExpression * body = outofline->queryChild(0);
+    assertex(body->getOperator()==no_embedbody);
+    ITypeInfo * returnType = funcdef->queryType()->queryChildType();
+
+    bool isImport = body->hasAttribute(importAtom);
+    IHqlExpression * formals = funcdef->queryChild(1);
+    assertex(formals->numChildren() == expr->numChildren());  // MORE - do default params change this?
+    unsigned flags = isImport ? EFimport : EFembed;
+    if (formals->numChildren()==0)
+        flags |= EFnoparams;
+    if (returnType->getTypeCode()==type_void)
+        flags |= EFnoreturn;
+
+    StringBuffer optionsStr;
+    ForEachChild(idx, body)
+    {
+        IHqlExpression *child = body->queryChild(idx);
+        if (child->isAttribute() && !isInternalEmbedAttr(child->queryName()))
+        {
+            if (optionsStr.length())
+                optionsStr.append(",");
+            optionsStr.append(child->queryName());
+            IHqlExpression * value = child->queryChild(0);
+            if (value)
+            {
+                optionsStr.append("=");
+                value->queryValue()->getUTF8Value(optionsStr);
+            }
+        }
+    }
+    IHqlExpression *languageAttr = body->queryAttribute(languageAtom);
+    HqlExprArray noParams;
+    OwnedHqlExpr langLoadCall = createTranslatedExternalCall(NULL, languageAttr->queryChild(0), noParams);
+    IValue *plugin = foldExternalCall(langLoadCall, foldOptions, templateContext);
+    if (plugin == nullptr)
+        return NULL;
+
+    Owned<IEmbedContext> __plugin = (IEmbedContext *) plugin->getIntValue();  // We declared as int since ecl has no opinter type - not sure what the clean fix is here...
+    DummyContext dummyContext;
+    Owned<IEmbedFunctionContext> __ctx = __plugin->createFunctionContextEx(&dummyContext,flags,optionsStr.str());
+
+    if (!isImport)
+    {
+        IValue *query = body->queryChild(0)->queryValue();
+        assertex(query);
+        StringBuffer queryText;
+        query->getUTF8Value(queryText);
+        __ctx->compileEmbeddedScript(queryText.lengthUtf8(), queryText.str());
+    }
+
+    // process all the parameters passed in
+    unsigned numParam = expr->numChildren();
+    for(unsigned i = 0; i < numParam; i++)
+    {
+        IHqlExpression * curParam = expr->queryChild(i);            //NB: Already folded...
+        IHqlExpression * curArg = formals->queryChild(i);
+        assertex(curArg != NULL);
+        ITypeInfo * paramType = curArg->queryType();
+        assertex(paramType != NULL);
+        IValue * paramValue = curParam->queryValue();
+        assertex(curParam->isConstant());
+        unsigned paramSize = paramType->getSize();
+        IIdAtom * paramId = curArg->queryId();
+        const char * name = str(paramId);
+        // if (!foldOptions & HFOpreserveCaseExternalParameter)  // Does not seem to be reliably passed in - is there another way ?
```

Comment:
I think we should always preserving case on external parameter names.  (This code is missing some brackets, but should probably be deleted).